Configuration/UTMAppleConfigurationDrive.swift

```diff
@@ -92,12 +104,22 @@ struct UTMAppleConfigurationDrive: UTMConfigurationDrive {
             try container.encodeIfPresent(imageName, forKey: .imageName)
         }
         try container.encode(isReadOnly, forKey: .isReadOnly)
+        try container.encode(isSparse, forKey: .isSparse)
         try container.encode(id, forKey: .identifier)
     }
     
     func vzDiskImage() throws -> VZDiskImageStorageDeviceAttachment? {
         if let imageURL = imageURL {
-            return try VZDiskImageStorageDeviceAttachment(url: imageURL, readOnly: isReadOnly)
+            if #available(macOS 12, *) {
+                /*
+                 * virtual disk cache mode have bugs,
+                 * when it is enabled or set to auto (default value)
+                 * may cause linux file system corrputed, especially in the case of heavy IO loads
+                 */
+                return try VZDiskImageStorageDeviceAttachment(url: imageURL, readOnly: isReadOnly, cachingMode:VZDiskImageCachingMode.uncached, synchronizationMode: VZDiskImageSynchronizationMode.full)
```

Comment:
Now, #5919 is updated to use .cached virtio drives for VMs, as it has been tested to avoid the filesystem problem:  https://github.com/utmapp/UTM/issues/4840#issuecomment-1824340975. The uncached option is confirmed to still lead to filesystem corruptions; therefore, it is recommended to remove this option here.

Reply:
OK